### PR TITLE
Add RemoveTx RPC endpoint

### PR DESCRIPTION
**Pull request by allanperlee**
Introduced light, experimental change to issue #357 . Will explore suggestions, if any. Thank you.

### CODE REVIEW

File: rpc/client/client.go [hunk 1]
```diff
@@ -552,8 +552,10 @@ func (c *Client) Tx(ctx context.Context, hash tmbytes.HexBytes, prove bool) (*ct
 }
 
 func (c *Client) RemoveTx(ctx context.Context, key types.TxKey) error {
-	//TODO implement me
-	panic("implement me")
+	if err := c.node.Mempool.RemoveTxByKey(key); err != nil {
+		fmt.Errorf("failed to remove transaction from mempool: %w", err)
```
[the remark this answers is not in the document]
Comment (allanperlee) [reply rpc/client/client.go:556]: Yes, sir. Let me submit the changes momentarily

**Review by tzdybal — CHANGES_REQUESTED** [review]
Please add a test to `rpc/client/client_test.go`.
Hint 1: you can take a look at `TestCheckTx`, and create something similar, but with extra step of removing Tx.
Hint 2: There is `mempool.TxKey` function to get a key for Tx.

**Comment by allanperlee** [comment]
@tzdybal I cannot find the `mempool.TxKey` method in the node directory, where can I find it? I want to learn if it takes parameters, and if a `node` is needed to use this method like in line 864.

**Comment by tzdybal** [comment]
`mempool` is a package `github.com/celestiaorg/optimint/mempool/` (you probably need to import it). 

https://github.com/celestiaorg/optimint/blob/d0dcc50eb69b3d206f0be8d56154a0a36eee4ee3/mempool/clist_mempool.go#L756

**Comment by allanperlee** [comment]
The package was imported and `TxKey` was used, but there is an error, stating that TxKey is not declared. Here is how I used it:

**Comment by tzdybal** [comment]
Ah this was moved to different place. In type `Tx` in `"github.com/tendermint/tendermint/abci/types"`, there is a method `Key`.

**Comment by allanperlee** [comment]
Yes, I committed the unit test. It fails with the wrong tx hash. It doesn't pass yet. I could not find the `Key` method in `github.com/tendermint/tendermint/abci/types` though. I chose the `TxKey`'s hashing instead provisionally. I found a method `Key` in `https://github.com/tendermint/tendermint/blob/master/types/tx.go#L23` , but the overriding or implementation of this interface is not found.

**Comment by nashqueue** [comment]
Close stale PR cc: @tzdybal , Our client does not have a RemoveTx anymore so there is no Endpoint to be added.

**Pull request closed without merging**